cloudify_ansible_sdk/__init__.py

```diff
@@ -0,0 +1,308 @@
+                 site_yaml_path,
+                 sources='localhost,',
+                 inventory_config=None,
+                 variable_manager_config=None,
+                 options_config=None,
+                 passwords=None,
+                 modules=None,
+                 private_key_file=None,
+                 run_data=None,
+                 verbosity=2):
+
+        self.display = Display(verbosity=verbosity)
+
+        self.loader = DataLoader()
+        self.modules = modules or []
+        self.passwords = passwords or {'conn_pass': None, 'become_pass': None}
+        self.private_key_file = private_key_file
+        self.sources = sources
+        self.site_yaml_path = site_yaml_path
+        self.verbosity = verbosity
+
+        # Configurations
+        self.inventory_config = inventory_config or {}
+        self.variable_manager_config = variable_manager_config or {}
+        self.options_config = options_config or {}
+
+        self.run_data = run_data or {}
+
+        self.inventory = self._set_inventory()
+        self.variable_manager = self._set_variable_manager()
+        self.options = self._set_options()
+        self.display.verbosity = self.options.verbosity
+
+        self.tqm = self._set_task_manager()
+        self.runner = self._set_runner()
+
+    def _set_inventory(self):
+        _kwargs = deepcopy(self.inventory_config)
+        if not isinstance(_kwargs, dict):
+            raise CloudifyAnsibleSDKError(
+                'inventory_config must be a dictionary.')
+        if 'loader' not in _kwargs:
+            _kwargs.update({'loader': self.loader})
+        if 'sources' not in _kwargs:
+            _kwargs.update({'sources': self.sources})
+        return InventoryManager(**_kwargs)
+
+    def _set_variable_manager(self):
+        _kwargs = deepcopy(self.variable_manager_config)
+        if not isinstance(_kwargs, dict):
+            raise CloudifyAnsibleSDKError(
+                'variable_manager_config must be a dictionary.')
+        if 'loader' not in _kwargs:
+            _kwargs.update({'loader': self.loader})
+        if 'inventory' not in _kwargs:
+            _kwargs.update({'inventory': self.inventory})
+        variable_manager = VariableManager(**_kwargs)
+        variable_manager.extra_vars = self.run_data
+        variable_manager.set_inventory(self.inventory)
+        return variable_manager
+
+    # A lot of this stuff I did not have time to validate.
+    def _set_options(self):
+        _kwargs = deepcopy(self.options_config)
+        if not isinstance(_kwargs, dict):
+            raise CloudifyAnsibleSDKError(
+                'options_config must be a dictionary.')
+        if 'ask_pass' not in _kwargs:
+            _kwargs['ask_pass'] = False
+        if 'ask_su_pass' not in _kwargs:
+            _kwargs['ask_su_pass'] = False
+        if 'ask_sudo_pass' not in _kwargs:
+            _kwargs['ask_sudo_pass'] = False
+        if 'ask_vault_pass' not in _kwargs:
```

Comment:
@EarthmanT In instead of having all of these if statement cannot we just do the following:
1. Define a dict with all possible kyes for option_config as  `OPTION_CONFIG_DEFAULT`
```
OPTION_CONFIG_DEFAULT = {
'ask_pass': False,
'become_user': 'root',
'extra_vars': []
....
}
```

2. Remove all if statements and replace them with the following code:
  ```
  for key, value in  OPTION_CONFIG.iteritems():
    if key not in  _kwargs:
       _kwargs[key] = value

  ```

I think that would be more readable

Reply:
done